src/DIRAC/WorkloadManagementSystem/Agent/JobAgent.py

```diff
@@ -694,7 +694,8 @@ def _checkSubmittedJobs(self):
         submissionErrors = []
         payloadErrors = []
         originalJobID = self.jobReport.jobID
-        for jobID, taskID in self.submissionDict.items():
+        for jobID in list(self.submissionDict.keys()):
+            taskID = self.submissionDict[jobID]
```

Comment:
Aren't the 2 exactly the same?

Reply:
Yes and no.
If I am not wrong, we should not delete keys from `self.submissionDict` while iterating over it as it can lead to unpredictable behavior and runtime errors.
This is why I am iterating over a copy of `self.submissionDict.keys()`.

Reply:
add this comment in the code please!